components/gallery/GalleryItemToolBar.vue

```diff
@@ -0,0 +1,106 @@
+    <NeoTooltip :label="$t('reload')" position="top">
+      <a no-shadow @click="handleReloadClick">
+        <NeoIcon
+          :icon="isLoading ? 'spinner-third' : 'arrow-rotate-left'"
+          size="medium"
+          :label="$t('reload')"
+          :spin="isLoading" />
+      </a>
+    </NeoTooltip>
+    <NeoTooltip :label="$t('fullscreen')" position="top">
+      <a no-shadow @click="$emit('toggle')">
+        <NeoIcon
+          icon="arrow-up-right-and-arrow-down-left-from-center"
+          size="medium" />
+      </a>
+    </NeoTooltip>
+    <NeoTooltip :label="$t('newTab')" position="top">
+      <a no-shadow @click="handleNewTab">
+        <NeoIcon icon="arrow-up-right" size="medium" />
+      </a>
+    </NeoTooltip>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { NeoIcon, NeoTooltip } from '@kodadot1/brick'
+
+import { useGalleryItem } from './useGalleryItem'
+import { MediaType } from '@/components/rmrk/types'
+import {
+  determineElementType,
+  mediaTypeElementSelectors,
+  resolveMedia,
+} from '@/utils/gallery/media'
+
+type ReloadElement =
+  | HTMLIFrameElement
+  | HTMLVideoElement
+  | HTMLImageElement
+  | null
+
+defineEmits(['toggle'])
+
+const { nftAnimationMimeType, nftMimeType } = useGalleryItem()
+
+const isLoading = ref(false)
+
+const mediaAndImageType = computed(() => {
+  const animationMediaType = resolveMedia(nftAnimationMimeType.value)
+  const imageMediaType = resolveMedia(nftMimeType.value)
+  return { animationMediaType, imageMediaType }
+})
+
+const getElementSelector = ({ imageMediaType, animationMediaType }) => {
+  const elementType = determineElementType(animationMediaType, imageMediaType)
+  return mediaTypeElementSelectors[elementType]
+}
+
+const reloadElement = (selector: string) => {
+  setTimeout(() => {
+    isLoading.value = false
+    const element: ReloadElement = document.querySelector(selector)
+    if (!element) {
+      return
+    }
+    if (mediaTypeElementSelectors[MediaType.IMAGE] === selector) {
+      const timestamp = new Date().getTime()
+      const url = new URL(element.src)
+      url.searchParams.set('t', timestamp.toString())
+      element.src = url.toString()
+    } else {
+      element.src += ''
+    }
+  }, 500)
+}
+
+const handleReloadClick = () => {
+  isLoading.value = true
+  const { animationMediaType, imageMediaType } = mediaAndImageType.value
+
+  return reloadElement(
+    getElementSelector({ animationMediaType, imageMediaType }),
+  )
+}
+
+const openInNewTab = (selector: string, attribute: string = 'src') => {
+  const element = document.querySelector(selector)
+  if (element) {
+    const src = element.getAttribute(attribute)
+    if (src) {
+      window.open(src, '_blank')
+    }
+  }
+}
+
+const handleNewTab = () => {
+  const { animationMediaType, imageMediaType } = mediaAndImageType.value
+
+  return openInNewTab(
+    getElementSelector({ animationMediaType, imageMediaType }),
+  )
+}
```

Comment:
`MediaType` may be `undefined`

Check the fallback code here:
![image](https://github.com/kodadot/nft-gallery/assets/31397967/84629d22-396d-4d94-bb21-c8280d52d4c6)

Reply:
I have added this back